pkg/network/tracer/tracer_linux_test.go

```diff
@@ -2547,6 +2552,181 @@ func setupDropTrafficRule(tb testing.TB) (ns string) {
 	return
 }
 
+func (s *TracerSuite) TestTLSClassification() {
+	t := s.T()
+	cfg := testConfig()
+
+	if !kprobe.ClassificationSupported(cfg) {
+		t.Skip("protocol classification not supported")
+	}
+	port, err := tracertestutil.GetFreePort()
+	require.NoError(t, err)
+	portAsString := strconv.Itoa(int(port))
+
+	tr := setupTracer(t, cfg)
+
+	type tlsTest struct {
+		name            string
+		postTracerSetup func(t *testing.T)
+		validation      func(t *testing.T, tr *Tracer)
+	}
+	tests := make([]tlsTest, 0)
+	for _, scenario := range []uint16{tls.VersionTLS10, tls.VersionTLS11, tls.VersionTLS12, tls.VersionTLS13} {
+		scenario := scenario
+		tests = append(tests, tlsTest{
+			name: strings.Replace(tls.VersionName(scenario), " ", "-", 1),
+			postTracerSetup: func(t *testing.T) {
+				srv := usmtestutil.NewTLSServerWithSpecificVersion("localhost:"+portAsString, func(conn net.Conn) {
```

Comment:
You could just specify a `0` port here and then retrieve the actual port the server is listening at to use later for the client.

Reply:
added